src/fmpz_mat/is_spd.c

```diff
@@ -0,0 +1,49 @@
+/*
+    Copyright (C) 2023 Jean Kieffer
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "fmpz.h"
+#include "fmpz_mat.h"
+
+int
+fmpz_mat_is_spd(const fmpz_mat_t A)
+{
+    slong d = fmpz_mat_nrows(A);
+    fmpz_mat_t tp, w;
+    fmpz_t det;
+    slong k;
+    int res = 1;
+
+    if (fmpz_mat_ncols(A) != d)
+    {
+        return 0;
+    }
+
+    fmpz_mat_init(tp, d, d);
+    fmpz_init(det);
+
+    fmpz_mat_transpose(tp, A);
+    if (!fmpz_mat_equal(tp, A))
+    {
+        res = 0;
```

Comment:
May be worth ensuring that test code covers this branch.

Reply:
I've tried to address this in https://github.com/flintlib/flint/pull/1524/commits/5840f9adb5cca6164d191d4678a1648e6bc39b1d. In general is there a good way of checking that a new contribution has a problem with test coverage?

Reply:
Sure, check the test coverage reports by codecov in the PR. (Or locally on your machine, ``./configure --enable-coverage && make -j check && make coverage``.)